crates/executor/src/executor/mod.rs

```diff
@@ -103,6 +117,18 @@ where
             tx_len = transactions.len(),
         );
 
+        // The withdrawals root on OP Stack chains, after Canyon activation, is always the empty
+        // root hash.
+        let mut withdrawals_root = self
+            .config
+            .is_canyon_active(payload.payload_attributes.timestamp)
+            .then_some(EMPTY_ROOT_HASH);
+        // If the Isthmus hardfork is active, the withdrawals root is the L2 to L1 message passer
+        // account.
+        if self.config.is_isthmus_active(payload.payload_attributes.timestamp) {
+            withdrawals_root = Some(self.message_passer_account()?);
+        }
```

Comment:
This always needs to go after the state root is sealed up, post-execution. Code looks good though, just need to move it back to where it used to be.

Reply:
The issue is an exclusive mutable reference to the trie db is given to the `State` through the builder earlier in the method (right below this added section) - see `&mut self.trie_db` which is needed to get the message passer account for the withdrawals root.